src/lib.rs

```diff
@@ -1,31 +1,40 @@
 //! SputnikVM implementation, traits and structs.
 //!
-//! SputnikVM work on two different levels. It can directly handles a
-//! transaction, or a normal Etheruem execution context. To interact
-//! with the virtual machine, you usually only need to work with the
-//! [VM](trait.VM.html) trait.
+//! SputnikVM works on two different levels. It can directly handle
+//! either a transaction or a normal Ethereum execution context. To
+//! interact with the virtual machine, you usually only need to work
+//! with the [VM](trait.VM.html) trait.
 //!
 //! ### Lifecycle
 //!
-//! A VM can be started given a `Transaction` (or `Context`) and a
-//! `BlockHeader`. The user can then `fire` or `step` to run it.
-//! `fire` runs through all possible steps until it cannot continue,
-//! while `step` only runs one step. Those functions would only fail
-//! if it needs some information (accounts in the current block, or
-//! block hashes of previous blocks) provided by `RequireError`. If
-//! this happens, one can use the function `commit_account` and
-//! `commit_blockhash` to commit those information to the VM, and
-//! `fire` or `step` again until it succeeds. The current VM status
-//! can always be obtained using the `status` function.
+//! A VM can be started after it is given a `Transaction` (or
+//! `Context`) and a `BlockHeader`. The user can then `fire` or `step`
+//! to run it.  `fire` runs the EVM code (given in field `code` of the
+//! transaction) until it finishes or cannot continue. However `step`
+//! only runs at most one instruction. If the virtual machine needs
+//! some information (accounts in the current block, or block hashes
+//! of previous blocks) it fails, returning a `RequireError`
+//! enumeration. With the data returned in the `RequireError`
+//! enumeration, one can use the function `commit_account` and
+//! `commit_blockhash` to commit the information to the VM. `fire` or
+//! `step` can be subsequently called to restart from that
+//! point. The current VM status can always be obtained using the
+//! `status` function.
 //!
 //! ### Patch
 //!
 //! Every VM is associated with a `Patch`. This patch tells the VM
 //! which Ethereum network and which hard fork it is on. You will need
 //! to specify the patch as the type parameter. To interact with
-//! multiple Patches at the same time, it is recommended that you use
+//! multiple patches at the same time, it is recommended that you use
 //! trait objects.
 //!
+//! The example below creates a new SputnikVM and stores it in
+//! variable `vm`. To do this, it must first create a
```

Comment:
I like this addition; it may be counterintuitive for many people that "a machine" would simply be stored in a single variable ;)